ReactWindows/ReactNative/Modules/Location/LocationModule.cs

```diff
@@ -0,0 +1,194 @@
+﻿using Newtonsoft.Json.Linq;
+using ReactNative.Bridge;
+using ReactNative.Collections;
+using ReactNative.Modules.Core;
+using System;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Windows.Devices.Geolocation;
+using Windows.Foundation;
+
+namespace ReactNative.Modules.Location
+{
+    class LocationModule : ReactContextNativeModuleBase
+    {
+        private IDisposable _geolocatorSubscription;
+
+        public LocationModule(ReactContext reactContext) 
+            : base(reactContext)
+        {
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "LocationObserver";
+            }
+        }
+
+        [ReactMethod]
+        public async void getCurrentPosition(
+            JObject options,
+            ICallback success,
+            ICallback error)
+        {
+            var locationOptions = LocationOptions.FromJson(options);
+
+            var geolocator = new Geolocator
+            {
+                DesiredAccuracy = locationOptions.HighAccuracy ? PositionAccuracy.High : PositionAccuracy.Default,
+                MovementThreshold = locationOptions.DistanceFilter,
+            };
+
+            try
+            {
+                var task = geolocator.GetGeopositionAsync().AsTask();
+                var completedTask = await Task.WhenAny(
```

Comment:
We're only trying to retrieve a single value in this ReactMethod, so an Observable is probably overkill.

---
In reply to: [70992474](https://github.com/ReactWindows/react-native-windows/pull/526#discussion_r70992474) [](ancestors = 70992474)